src/Microsoft.AspNetCore.OData/Batch/ODataBatchReaderExtensions.cs

```diff
@@ -236,5 +268,103 @@ private static HttpContext CreateHttpContext(HttpContext originalContext)
 
             return context;
         }
+
+        /// <summary>
+        /// Extract preferences that can be inherited from the overall batch request to
+        /// an individual request.
+        /// </summary>
+        /// <param name="batchPreferences">The value of the Prefer header from the batch request</param>
+        /// <returns>comma-separated preferences that can be passed down to an individual request</returns>
+        private static string GetPreferencesToInheritFromBatch(string batchPreferences)
+        {
+            // do not inherit respond-async and continue-on-error (odata.continue-on-error in OData 4.0)
+            var preferencesToIgnore = new string[] { "respond-async", "continue-on-error", "odata.continue-on-error" };
+            var preferencesToInherit = batchPreferences.SplitPreferences()
+                .Where(value => 
+                !preferencesToIgnore.Any(
+                    prefToIgnore => value.Trim().ToLowerInvariant().StartsWith(prefToIgnore)))
+                .Select(value => value.Trim());
+            return string.Join(", ", preferencesToInherit);
+        }
+
+        /// <summary>
+        /// Merges the preferences from the batch request and an individual request inside the batch into one value.
+        /// If a given preference is defined in both the batch and individual request, the one from the individual
+        /// request is retained and the one from the batch is discarded.
+        /// </summary>
+        /// <param name="individualPreferences">The value of the Prefer header from the individual request inside the batch</param>
+        /// <param name="batchPreferences">The value of the Prefer header from the overall batch request</param>
+        /// <returns>Value containing the combined preferences</returns>
+        private static string MergeIndividualAndBatchPreferences(string individualPreferences, string batchPreferences)
+        {
+            if (string.IsNullOrEmpty(individualPreferences))
+            {
+                return batchPreferences;
+            }
```

Comment:
```suggestion
            }
            
``` Newline after closing bracket...